torch/_prims_common/__init__.py

```diff
@@ -168,27 +173,27 @@ def compare_tensor_meta(
             pass
         else:
             msg = f"Devices {a.device} and {b.device} are not equal!"
-            raise AssertionError(msg)
+            raise TensorCrossRefException(msg)
 
     # Stride checking is currently disabled, see https://github.com/pytorch/pytorch/issues/78050
     if check_strides:
         same_strides, idx = check_significant_strides(a, b)
         if not same_strides:
             msg = f"Stride mismatch! Strides are {a.stride()} and {b.stride()} (mismatched at {idx})!"
-            raise RuntimeError(msg)
+            raise TensorCrossRefException(msg)
 
         if a.storage_offset() != b.storage_offset():
             msg = f"Storage offset mismatch! Storage offsets are {a.storage_offset()} and {b.storage_offset()}!"
-            raise RuntimeError(msg)
+            raise TensorCrossRefException(msg)
 
     if check_conj:
         if a.is_conj() != b.is_conj():
-            raise RuntimeError(
+            raise TensorCrossRefException(
```

Comment:
nit: call it FakeCrossRef?

Reply:
is it ok to keep it like this? I'm raising this in `_prims_common.py/compare_tensor_meta()` too, which isn't necessarily for FakeTensors

Reply:
Ah I didn't realize. TensorMetadataMismatchError maybe?

Reply:
Or maybe just MetadataMismatchError if we didn't want Tensor in the name

Reply:
thanks, just changed!